### PR TITLE
[Merged by Bors] - feat: the category of short complexes is preadditive

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Homology.ShortComplex.Homology
+import Mathlib.CategoryTheory.Preadditive.AdditiveFunctor
+
+/-! Homology of preadditive categories
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean:9]:
```suggestion
/-!
# Homology of preadditive categories
```

File: Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+TODO: Introduce the notion of homotopy of morphisms of short complexes.
+
+-/
+
+namespace CategoryTheory
+
+open Category Limits Preadditive
+
+variable {C : Type _} [Category C] [Preadditive C]
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean:22]:
```suggestion
variable {C : Type*} [Category C] [Preadditive C]
```

File: Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+section LeftHomology
+
+variable {φ φ' : S₁ ⟶ S₂} {h₁ : S₁.LeftHomologyData} {h₂ : S₂.LeftHomologyData}
+
+namespace LeftHomologyMapData
+
+variable (γ : LeftHomologyMapData φ h₁ h₂) (γ' : LeftHomologyMapData φ' h₁ h₂)
+
+/-- Given a left homology map data for morphism `φ`, this is induced left homology
+map data for `φ`. -/
```
Comment (TwoFX) [inline Mathlib/Algebra/Homology/ShortComplex/Preadditive.lean:77]:
```suggestion
/-- Given a left homology map data for morphism `φ`, this is induced left homology
map data for `-φ`. -/
```

**Review by TwoFX — APPROVED** [review]
Thanks!
bors d+

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**